### PR TITLE
[ylylma] iP

### PR DESCRIPTION
**Pull request by ylyma**
# Kiera

> "Your mind is for having ideas, not holding them." -David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Kiera frees your mind of having to remember things you need to do. With Kiera, you can effortlessly keep track of your tasks and stay organized like never before.

🚀 **Features:**
- [X] 📝 Managing tasks
- [X] 🔎 Search and filter by date or keyword
- [ ] ⏰ Reminders (coming soon)

💻 **Getting Started**
1. Download the .jar file from [here](https://github.com/ylyma/ip/releases/tag/Level-9)
2. Double-click it
3. Add your tasks!

📣 **Usage**
* Add tasks
   ```
   todo task1
   deadline task2 /by 2023-09-04
   event task3 /from 2023-09-04 1300 /to 1500
   ```
* List tasks
   ```
   list
   ```
* Delete tasks by number
  ```
  delete 2
  ```
* Find tasks by keyword
  ```
  find task
  ```
* Filter tasks by date
  ```
  deadline-date 2023-09-04
  deadline-today
  event-date 2023-09-04
  event-today
  ```
-----
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/kiera/command/DeleteCommand.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package kiera.command;
+
+import kiera.Storage;
+import kiera.TaskList;
+import kiera.Ui;
+import kiera.exception.KieraException;
+import kiera.task.Task;
+
+/**
+ * Represents a command for deleting a tasl from tha taskList.
```
Comment (RSXIX) [inline src/main/java/kiera/command/DeleteCommand.java]: Think you mis spelled task here

File: src/main/java/kiera/command/ListCommand.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (tasks.isEmpty()) {
+            System.out.println("nothing for you to do yet!");
+        }
+        String result = tasks.getTasks().stream()
```
Comment (RSXIX) [inline src/main/java/kiera/command/ListCommand.java:24]: Neat use of streams

**Review by RSXIX — COMMENTED** [review]
The code felt very succient and well thought out. Good use of readable variables and little to none violation of coding standards.

File: src/main/java/kiera/Kiera.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+//                c.execute(tasks, ui, storage);
+//                isExit = c.isExit();
+//            } catch (KieraException e) {
+//                ui.showError(e.toString());
+//            } finally {
+//                ui.showLine();
+//            }
+//        }
+//    }
+
```
Comment (Nauxe) [inline src/main/java/kiera/Kiera.java]: I think you can remove this section of code or stash it locally

File: src/main/java/kiera/Parser.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        if (input.startsWith("deadline-date")) {
+            String desc = input.replace("deadline-date ", "");
+            return new FilterDateCommand(TaskType.DEADLINE, desc);
+        } else if (input.startsWith("event-date")) {
+            String desc = input.replace("event-date ", "");
+            return new FilterDateCommand(TaskType.EVENT, desc);
+        }
+
+        if (input.startsWith("deadline-today")) {
+            String desc = LocalDate.now().toString();
```
Comment (Nauxe) [inline src/main/java/kiera/Parser.java:77]: Good idea to let users create deadlines that are due on the current day!

**Review by Nauxe — COMMENTED** [review]
LGTM! Overall very clean code! I agree with the other review about using stream and filter, which makes the code look extremely neat. Your bot also has a really cute personality! ^-^

**Pull request closed without merging**